tls/s2n_security_policies.c

```diff
@@ -636,6 +636,11 @@ int s2n_security_policies_init()
         const struct s2n_ecc_preferences *ecc_preference = security_policy->ecc_preferences;
         notnull_check(ecc_preference);
         GUARD(s2n_check_ecc_preferences_curves_list(ecc_preference));
+        if (security_policy != &security_policy_null) {
+            /* catch any offending security policy that does not support P-256 */
+            GUARD_AS_POSIX(s2n_ecc_preferences_includes_p256(ecc_preference));
```

Comment:
Why does this require a separate method? Why not just use s2n_ecc_preferences_includes_curve directly?

Reply:
And why do this is production code instead of in a unit test that iterates over all security policies?

Reply:
> Why does this require a separate method? Why not just use s2n_ecc_preferences_includes_curve directly?

I can use s2n_ecc_preferences_includes_curve() if that's preferred

> And why do this is production code instead of in a unit test that iterates over all security policies?

1. this is where all security policies is in scope
2. this is inline with checks done in this security_policies block. even though it has a one time initialization cost during runtime, but a violation of this rule will also be caught during unit tests in this function